### ISSUE TITLE
Provide Events as Flow streams

### ISSUE THREAD

**Issue by rossdanderson** [rjaros/kvision#140]
It's quite nice to be able to subscribe to event flows on widgets for things like input and clicks, since you can easily use all the Flow operators (debounce, combine, merge etc.), and merge it with other non-ui data.

Currently I am using extension methods for this, but this means that it suffers the existing issue where a later collector of the flow will detach the first since I cannot guarantee a broadcast channel is shared underneath (at least without some kind of global state).

Here are my extension examples:
```
/**
 * Note that this can only be collected once at a time, a second collector will detatch the first
 */
inline val <reified T : Widget> T.onInput: Flow<T>
    get() = callbackFlow {
        onEvent {
            input = {
                offer(self)
            }
        }
        awaitClose {
            onEvent { input = {} }
        }
    }

/**
 * Note that this can only be collected once at a time, a second collector will detatch the first
 */
inline val <reified T : Button> T.onClick: Flow<T>
    get() = callbackFlow {
        onEvent {
            click = {
                offer(self)
            }
        }
        awaitClose {
            onEvent { click = {} }
        }
    }
```

And here is a simple usecase:
```
class BookPanel(
    private val scope: CoroutineScope
) : VPanel() {
    init {
        border = Border(1.px, BorderStyle.SOLID, Col.SILVER)
        width = 500.px

        val baseProductText = text(label = "Base product")
        val quoteProductText = text(label = "Quote product")

        val requestButtonDisabledFlow = baseProductText.onInput
            .combine(quoteProductText.onInput) { baseProduct, quoteProduct ->
                baseProduct.value.isNullOrEmpty() || quoteProduct.value.isNullOrEmpty()
            }
            .debounce(100)

        button("Request book") {

            requestButtonDisabledFlow.onEach { disabled = it }.launchIn(scope)

            onClick
                .onEach {
...
                }
                .launchIn(scope)
        }
    }
}
```

**Comment by rjaros** [comment]
I've made some major enhancements to the event handling code in KVision. Multiple event listeners will be fully supported and also removing single listeners will be possible. 
This will allow to create flows like yours without issues. 

Since I don't want to add coroutines dependency to the main module, I've created a new module - "kvision-event-flow" and defined there a few, ready to use flow functions: https://github.com/rjaros/kvision/blob/master/kvision-modules/kvision-event-flow/src/main/kotlin/pl/treksoft/kvision/event/EventFlow.kt

All these changes will be published in KVision 3.0.0 soon.
